libr/anal/p/anal_x86_cs.c

```diff
@@ -1136,7 +1136,15 @@ static void anop_esil (RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len
 		{
 			char *src = getarg (&gop, 1, 0, NULL);
 			char *dst = getarg (&gop, 0, 1, "&");
+			char *p;
 			esilprintf (op, "%s,%s,$0,of,=,$0,cf,=,$z,zf,=,$s,sf,=,$o,pf,=", src, dst);
+			if ((a->bits == 64) && (p = strchr (dst, (int)','))) {
+				char *dst64;
+				*p = '\0';
+				if (!(dst64 = r_reg_32_to_64 (a->reg, dst)))
```

Comment:
what about if () { r_strbuf_append} to avoid the memleak and make the code more readable?